new-log-viewer/src/utils/config.ts

```diff
@@ -0,0 +1,128 @@
+import {PAGE_SIZE} from "../contexts/StateContextProvider";
+import {
+    CONFIG_NAME,
+    ConfigMap,
+    ConfigUpdate,
+    LOCAL_STORAGE_KEY,
+} from "../typings/config";
+
+
+const DECODER_OPTIONS_DEFAULT = {
+    formatString: "%d{yyyy-MM-dd HH:mm:ss.SSS} [%process.thread.name] %log.level" + " %message%n",
+    logLevelKey: "log.level",
+    timestampKey: "@timestamp",
+};
+const THEMES = ["light",
+    "dark"];
+const THEME_DEFAULT = "light";
+
+/**
+ * Retrieves the configuration value from local storage based on the provided configuration name.
+ *
+ * @template T - The type of the configuration name.
+ * @param code The configuration name to retrieve the value for.
+ * @return - The configuration value or null if not found.
+ */
+const getConfig = <T extends CONFIG_NAME>(code: T): ConfigMap[T] | null => {
+    switch (code) {
```

Comment:
What happens if `window.localStorage.getItem()` returns `null` because the value was never set?

We may create another function `initConfigs()` to init all configs when the React app starts, or preferrably - we can call `setConfig(defaulConfigValue)` whenever the value retrieved from localStorage is null and return `defaulConfigValue`.